sympy/solvers/tests/test_solvers.py

```diff
@@ -1372,12 +1372,16 @@ def test_issue_5901():
     assert solve([f(x) - 3*f(x).diff(x)], f(x)) == \
         {f(x): 3*D}
     assert solve([f(x) - 3*f(x).diff(x), f(x)**2 - y + 4], f(x), y) == \
-        [{f(x): 3*D, y: 9*D**2 + 4}]
-    assert solve(-f(a)**2*g(a)**2 + f(a)**2*h(a)**2 + g(a).diff(a),
-                h(a), g(a), set=True) == \
-        ([g(a)], {
-        (-sqrt(h(a)**2*f(a)**2 + G)/f(a),),
-        (sqrt(h(a)**2*f(a)**2+ G)/f(a),)})
+        [(3*D, 9*D**2 + 4)]
+    eq = -f(a)**2*g(a)**2 + f(a)**2*h(a)**2 + g(a).diff(a)
+    syms = [g(a), h(a)]
+    syms_found, sols_found = solve(eq, h(a), g(a), set=True)
+    assert syms_found == syms
+    # For these expressions dummy_eq doesn't work so we extract the dummy
+    X1 = list(sols_found)[0][0]
+    sols = { (X1, +sqrt(X1**2*f(a)**2 - G)/f(a)),
+             (X1, -sqrt(X1**2*f(a)**2 - G)/f(a)) }
+    assert sols == sols_found
```

Comment:
Is there a change to these tests?

Reply:
> Is there a change to these tests?

What's the purpose of the dummy variable `_X1` here?

Reply:
I don't remember. Does the PR change the output of `dsolve` in these cases?

Reply:
> Does the PR change the output of `dsolve` in these cases?

No it doesn't. This is the original test. 

> https://github.com/sympy/sympy/blob/3572b1955d089e94d5af5b388b557fc26719ee1b/sympy/solvers/tests/test_solvers.py#L1376-L1380
 
In the PR, one variable `h(a)` for some reason, becomes a dummy variable.

Reply:
Seems like the real culprit is the behaviour of solve. What does `solve(f(x,y), x, y)` give?

Reply:
I think this should best be left out for another PR, where I hope to modify the behaviour of `solve()` a little bit. 
```
>>> solve(x**2-y**2,x,y)
[(-y, y), (y, y)]]
>>> solve(x - y**2, x, y)
[(y**2, y)]
```
`solve(f(x,y),x,y)` ought to return the solutions for both the variables `x` and `y`. This holds true for the first example but not for the second one.

Reply:
> In the PR, one variable `h(a)` for some reason, becomes a dummy variable.

I don't understand what this means. Does the PR change the output of `solve`?

Reply:
> > In the PR, one variable `h(a)` for some reason, becomes a dummy variable.
> 
> I don't understand what this means. Does the PR change the output of `solve`?

Yes it does, for this case.

Reply:
Since you want to change these tests, what I'd like to know is whether you want `solve` to return the output it returned [previously](https://github.com/sympy/sympy/blob/3572b1955d089e94d5af5b388b557fc26719ee1b/sympy/solvers/tests/test_solvers.py#L1376-L1380) or to keep these changes.

Reply:
I don't remember this from working on this before.

What causes the change in output?

Reply:
> 
> What causes the change in output?

Because the position of this block 
```
    if (
            not flags.get('dict', False) and
            solution and
            ordered_symbols and
            not isinstance(solution, dict) and
            all(isinstance(sol, dict) for sol in solution)
    ):
        solution = [tuple([r.get(s, s).subs(r) for s in symbols])
                    for r in solution]
```
was changed from [here](https://github.com/sympy/sympy/blob/fdb6e1f4add7f8522100ff35ebfe2c4db4e2a9ad/sympy/solvers/solvers.py#L1150) to [here](https://github.com/sympy/sympy/pull/20790/files#diff-d914c8b9dd231a57e757eb21513e3cff87b0aba6fbe3dc9ad128a3c16ad03e4bR1171)

Reply:
Maybe that shouldn't be changed then